### PR TITLE
Gpu Support

### PR DESCRIPTION
**Pull request by Andrew-Clappison**
New Classes: Fitting_Batch, CostsFixedPSD, 

GPU Support: Params (RWW2, EEG, BOLD), Par, BOLD, EEG, RWW2, mmRWW2, CostsFixedFC, CostsFixedPSD, CostsMean, CostsmmRWW2, Model_fitting, Fitting_FNGFPG, Fitting_Batch

GPU support added to sphinx docs example 002, with new example 005 also created

Extracted example 002 demo data into a function

### CODE REVIEW

**Comment by Andrew-Clappison** [comment]
This is related to Github Issues: #28, #84, #87, #116

File: whobpyt/optimization/cost_PSD.py [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 
 class CostsPSD():
+    '''
```
Comment (tmorshed) [inline whobpyt/optimization/cost_PSD.py:5]: Suggestion: Add a line to warn for the deprecation. a la [this link](https://dev.to/hckjck/python-deprecation-2mof#:~:text=Module%20deprecation,top%20level%20of%20that%20module).

For instance, define this before the class is initiated.
```python
from warnings import warn

def a_deprecated_function():
    warn('This method is deprecated.', DeprecationWarning, stacklevel=2)
```
Then adding this under `init`:
```python
warn(f'{self.__class__.__name__} will be deprecated.', DeprecationWarning, stacklevel=2)
```

**Comment by tmorshed** [comment]
Great work! Just made one suggestion @Andrew-Clappison. Should be good to merge otherwise. 👍🏻

File: whobpyt/optimization/cost_PSD.py (hunk 1 above)
Comment (Andrew-Clappison) [reply whobpyt/optimization/cost_PSD.py:5]: This function should be updated in the future, but message is okay for now.

**Review by JohnWangDataAnalyst — APPROVED** [review]

**Pull request merged**